tests/acceptance/features/apiSpaces/removeSpaces.feature

```diff
@@ -0,0 +1,181 @@
+    And user "Alice" has uploaded a file inside space "Owner deletes space" with content "some content" to "folderForDeleting/sub1/sub2/text.txt"
+    When user "Alice" removes a space "Owner deletes space"
+    Then the HTTP status code should be "200"
+    When user "Alice" lists all available spaces via the GraphApi
+    Then the json responded should not contain a space with name "Owner deletes space"
+
+#   editor role
+
+  Scenario: A editor can delete a folder with some subfolders in a Space via the webDav API
+    Given user "Alice" has created a space "Editor deletes folder" of type "project" with quota "10"
+    And user "Alice" has created a folder "folderForDeleting/sub1/sub2" in space "Editor deletes folder"
+    And user "Alice" has shared a space "Editor deletes folder" to user "Brian" with role "editor"
+    When user "Brian" removes the object "folderForDeleting" from space "Editor deletes folder"
+    Then the HTTP status code should be "200"
+    And for user "Brian" the space "Editor deletes folder" should not contain these entries:
+      | folderForDeleting |
+
+
+  Scenario: A editor can delete a subfolder in a Space via the webDav API
+    Given user "Alice" has created a space "Editor deletes subfolder" of type "project" with quota "10"
+    And user "Alice" has created a subfolder "folder/subFolderForDeleting" in space "Editor deletes subfolder"
+    And user "Alice" has shared a space "Editor deletes subfolder" to user "Brian" with role "editor"
+    When user "Brian" removes the object "folder/subFolderForDeleting" from space "Editor deletes subfolder"
+    Then the HTTP status code should be "200"
+    And for user "Brian" the space "Editor deletes subfolder" should contain these entries:
+      | folder |
+    And for user "Brian" folder "folder/" of the space "Editor deletes subfolder" should not contain these entries:
+      | subFolderForDeleting |
+
+
+  Scenario: A editor can delete a file in a Space via the webDav API
+    Given user "Alice" has created a space "Editor deletes file" of type "project" with quota "20"
+    And user "Alice" has uploaded a file inside space "Editor deletes file" with content "some content" to "text.txt"
+    And user "Alice" has shared a space "Editor deletes file" to user "Brian" with role "editor"
+    When user "Brian" removes the object "text.txt" from space "Editor deletes file"
+    Then the HTTP status code should be "200"
+    And for user "Brian" the space "Editor deletes file" should not contain these entries:
+      | text.txt |
+    When user "Brian" lists all available spaces via the GraphApi
+    Then the json responded should contain a space "Editor deletes file" with these key and value pairs:
+      | key          | value              |
+      | name         | Editor deletes file |
+      | quota@@@used | 0                  |
+
+
+  Scenario: A editor can delete a empty Space via the webDav API
```

Comment:
I am a bit surprised by this. But maybe this is the requirement?
I was thinking that spaces would have at least 3 roles like:
- Viewer - read-only of everything in the space
- Editor - can create/change/delete the resources (files and folders) in the space
- Manager - can manage access to the space (grant and revoke roles to users...) and can delete the whole space.

Reply:
AC in https://jira.owncloud.com/browse/OCIS-2063
viewer has no right to delete

Reply:
You are right, the editor should not be able to delete space.
https://jira.owncloud.com/browse/OCIS-2043
I will collect more information tomorrow and change some cases